cmd/rm.go

```diff
@@ -37,25 +34,49 @@ func rm(cmd *cobra.Command, args []string) (err error) {
 		return
 	}
 	if pathMetaData.Tag == folder {
-		return fmt.Errorf("rm: cannot remove ‘%s’: Is a directory", path)
+		if !recursive {
+			return fmt.Errorf("rm: cannot remove ‘%s’: Is a directory", path)
+		}
+		entries, err := ListFiles(path)
+		if err != nil {
+			return err
+		}
+		for _, e := range entries {
+			switch e.Tag {
+			case "folder":
+				removePath(pathmod.Join(path, e.Folder.Name), recursive)
```

Comment:
We needn't do this. The `Delete` API is recursive by default, see https://www.dropbox.com/developers/documentation/http/documentation#files-delete:

> If the path is a folder, all its contents will be deleted too.